Colcoction/unitys-workshop
# Consolidate Vertical Deck Parsing and Rendering

## Summary
- Adds a `CARD_FORM` configuration (values: deck, back, character). Backs don't rely on this and are left unconfigured for now.
- Adds a new method for parsing deck body text, `parseBodyText` in `common-after.js`.
- Adds a new method for rendering the parsed output of `parseBodyText, `drawCardBodyVerticalDeck`. As the name indicates, this is configured only for cards with the card form "deck" and the orientation "vertical".
- Unifies `effectPhaseFontSize` to depend on card form and orientation

## Test Plan
For hero and villain deck front pages:
1. Add reactions and powers with the following and verify it renders properly.
  a. No text
  b. 1 line of text
  c. multiple lines of text
2. Add all 5 phases and toggle high/low contrast. Verify that these all load correctly.
3. Declare a phase label in the following ways and verify they render properly.
  a. On its own line
  b. With "PHASE LABEL: some text".
  c. With "[PHASE Label] some text"
  d. With"<> phase LABEL", "|> pHaSE lAbEl", and "[] pHASE lABEL"
  e. With "[] phase label : some text".
4. Add some blocks of normal text with the following. Verify they load properly.
   a. A 2-word bolded term over a line break (it should break).
   b. A number and letter (e.g. "3 damage") over a line break (it should _not_ break).
   c. (can't do this for italicized terms since there's no user setting, but bold-testing should be enough)

For environment deck front, character front, and character back:
1. Add a phase label, a power, and some text. Verify that they render properly.

## Screenshots
![image](https://user-images.githubusercontent.com/8094985/233821153-202a1874-7831-42e0-b768-a2a568884049.png)
![image](https://user-images.githubusercontent.com/8094985/233821260-6204e6e2-7db6-41ac-858d-e5c12cb7bc75.png)
![image](https://user-images.githubusercontent.com/8094985/233821295-87f68be6-e88f-4b2b-ab95-fea722e7479d.png)
![image](https://user-images.githubusercontent.com/8094985/233821334-06b9ddd6-0666-4e71-968d-9f11838f0d53.png)

## Reviewers
@Colcoction 
@sjzhu
